test-suite/test_utils.c

```diff
@@ -0,0 +1,490 @@
+#include "test_utils.h"
+#define TEST_DEBUG 0
+#define BUF_SIZE 1024
+
+#if 0
+/**
+ * Prints mpibind_t imput parameters
+ **/
+static void print_test_params(mpibind_t *handle) {
+  printf(
+      "tasks: %d\t threads: %d\n greedy: %d\n gpu_optim: %d\n smt: %d\n "
+      "restr_type: %d\t restrict_ids %s",
+      handle->ntasks, handle->in_nthreads, handle->greedy, handle->gpu_optim,
+      handle->smt, handle->restr_type, handle->restr_set);
+}
+#endif
+
+/*
+ * mpibind relies on Core objects. If the topology
+ * doesn't have them, use an appropriate replacement.
+ * Make sure to always use get_core_type and get_core_depth
+ * instead of HWLOC_OBJ_CORE and its depth.
+ * Todo: In the future, I may need to have similar functions
+ * for NUMA domains.
+ */
+static int get_core_depth(hwloc_topology_t topo) {
+  return hwloc_get_type_or_below_depth(topo, HWLOC_OBJ_CORE);
+}
+
+/**
+ * Initialize a test struct to default values.
+ * This mimics the behavior of mpibind_init
+ * **/
+int mpibind_test_t_init(mpibind_test_t *hdl) {
+  if (hdl == NULL) {
+    return 1;
+  }
+
+  hdl->ntasks = 0;
```

Comment:
An mpibind handle is an opaque object. It should not be used directly. The structure behind it may change and should not break the test suite. 

```suggestion
  hdl->ntasks = 0;
```

Reply:
I replaced the described cases with the new getters. One thing to note is that I did make an almost identical struct called mpibind_test_t. The reason for this was so that I wouldn't be parsing test answers straight into the mpibind_t handles, and so that if needed, we could add members to the imitation struct to hold information that might be useful for current or future tests.